apps/client/app/routes/_layout.emails_.$emailId.tsx

```diff
@@ -0,0 +1,104 @@
+import { json, LoaderFunctionArgs } from "@remix-run/node";
+import { NavLink, useLoaderData, useNavigate } from "@remix-run/react";
+import { MOCKED_EMAILS } from "../../mocks/emails";
+import { ChevronLeftIcon, PencilIcon } from "@heroicons/react/24/solid";
+import { useState } from "react";
+import { render } from "@react-email/render";
+
+
+type EmailViewer = "CONTENT" | "HTML" | "CODE";
+
+async function getEmailById(id: string) {
+  await new Promise((resolve) => setTimeout(resolve, 500));
+  return MOCKED_EMAILS.find(email => email.id === id);
+}
+
+export const loader = async ({ params }: LoaderFunctionArgs) => {
+  if(params.emailId) {
+    const email = await getEmailById(params.emailId);
+    if (!email) {
+      throw new Response("Email not Found", { status: 404 });
+    }
+    return json({ email });
+  }
+};
+
+const EmailDetails = () => {
+  const { email } = useLoaderData<typeof loader>();
+  const [selectedView, setSelectedView] = useState<EmailViewer>("CONTENT");
+  const navigate = useNavigate();
+
+  const viewButtonClass = (view: EmailViewer) => {
+    return (
+        "px-3 py-1 rounded-lg " +
+      (view === selectedView ? "bg-black text-white" : "hover:bg-gray-200")
+    )
+  }
+
+  /*
+  const html = render(
+    <div dangerouslySetInnerHTML={ { __html: email.content } }/>,
+    {
+      pretty: true,
+    }
+  );
+  console.log(html);
+  */
```

Comment:
Is this needed or can be removed?

Reply:
Im currently testing some things to render the html in a formatted way so i left it for now.